contract/src/lib.rs

```diff
@@ -71,45 +66,41 @@ pub struct Answer {
     clue: String,
 }
 
-#[derive(BorshDeserialize, BorshSerialize, Deserialize, Serialize, Debug)]
-#[serde(crate = "near_sdk::serde")]
+#[near(serializers = [borsh, json])]
 pub enum PuzzleStatus {
     Unsolved,
-    Solved { solver_pk: PublicKey },
+    Solved { solver_pk: String },
     Claimed { memo: String },
 }
 
-#[derive(Serialize)]
-#[serde(crate = "near_sdk::serde")]
+#[near(serializers = [borsh, json])]
 pub struct UnsolvedPuzzles {
     puzzles: Vec<JsonPuzzle>,
     creator_account: AccountId,
 }
 
-#[derive(Serialize, Deserialize)]
-#[serde(crate = "near_sdk::serde")]
+#[near(serializers = [borsh, json])]
 pub struct JsonPuzzle {
     /// The human-readable public key that's the solution from the seed phrase
     solution_public_key: String,
     status: PuzzleStatus,
-    reward: Balance,
+    reward: NearToken,
     creator: AccountId,
     dimensions: CoordinatePair,
     answer: Vec<Answer>,
 }
 
-#[derive(BorshDeserialize, BorshSerialize, Debug)]
+#[near(serializers = [borsh])]
```

Comment:
Isn't is the default serialization behavior of `#[near]`?

Reply:
fixed